### ISSUE TITLE
Emailproxy.exe is being reported as a containing a virus by McAffe antivirus.

### ISSUE THREAD

**Issue by Rick-G1** [simonrob/email-oauth2-proxy#265]
Because of the common problem of many PyPI compilations being detected as a virus, such as Emailproxy.exe is, would it be possible to list the MD5 hash for at least the Windows version of the executable file emailproxy.exe.   This would give people downloading the package a degree of confidence that the downloaded files do not contain a virus.

This can be done in Windows with "certutil -hashfile  emailproxy.exe MD5"  and then publish the hash number.  (The number I have on the latest one I've downloaded [2024_05_25]  is   65dfebe8463511e0f8a69f6cc4f49b8f)

Thanks!!

**Comment by simonrob** [comment]
Thanks for the suggestion. It's a shame that virus scanners incorrectly do this, but always worth reporting these incorrect results to them to improve detection (different providers support this in different ways, but here's the [AVG reporting page](https://secure.avg.com/submit-sample), for example).

Re: MD5 hash – I'm not sure what this would achieve, given that the compiled version of the proxy is built [entirely automatically](https://github.com/simonrob/email-oauth2-proxy/blob/build/.github/workflows/pyinstaller.yml), and any person or process that has access to set edit the release page to add the hash could also maliciously edit the compiled file. Still, in case it helps, the hash from my side is also `65dfebe8463511e0f8a69f6cc4f49b8f`.

**Comment by Rick-G1** [comment]
Hi: 

 

First of all --- Thank you so much for your assistance !!  I really appreciate it !!  Thanks for reporting the hash value.

 

Tried to report this file to McAffe, but their reporting mechanism said that the 32MB file was "too big"…  I also found a rather interesting web site that checks about 75 different antivirus tools.  The site is  https://www.virustotal.com/gui/home/upload and you just drag the suspected file onto it.   It shows that emailproxy.exe shows problems on 14 of 75 tools.   

 

As for the hash…. If you change a single line of code (either in your python source or after compilation) the MD4 hash will be different, and you'd have to be very creative to then add "stuff" to malicious code to make the number the same.   So this gives people confidence in the fact that the data supplied is what the author intended. 

 

I’m going to try to use this for a windows machine and Outlook 2007 with OAuth2.   Apparently I could get the latest Outlook that has OAuth2 for free, but I've heard it's filled with ads and reports information back to Microsoft.    The old one just works.  Might be tempted to switch to Thunderbird, but calendars and contacts are different, so I'm hoping this creative package of yours will work for my situation. 

 

Thanks Again !!!

Rick Groome
